api/v1alpha4/types.go

```diff
@@ -420,6 +420,8 @@ type DataDisk struct {
 	// NameSuffix is the suffix to be appended to the machine name to generate the disk name.
 	// Each disk name will be in format <machineName>_<nameSuffix>.
 	NameSuffix string `json:"nameSuffix"`
+	// UltraSSDEnabled is set to true if Ultra disk is enabled for data disks
+	UltraSSDEnabled bool `json:"ultraSSDEnabled,omitempty"`
```

Comment:
now that I'm looking at this more closely, is there a reason we need this extra API field? why not use ManagedDisk. StorageAccountType? What's the expected value for a user to add a disk with ultra disk for StorageAccountType right now?

Reply:
sounds quiet right, didn't think of using StorageAccountType to confirm if ultra disk is enabled, we do not need this field.

Reply:
>  is it possible that ManagedDiskParameters is left empty as it is optional? In that case, would we miss the info about ultra ssd?

if ManagedDiskParameters is empty we just use the Azure defaulted storage type as before. Ultra ssd would be disabled in that case since it's an opt in feature. Does that make sense?

Reply:
sounds fair, will remove the extra field `UltraSSDEnabled`  added and will make use of `StorageAccountType`